### ISSUE TITLE
The `@stylistic/less/unit-case` rule recases text Less compiles as a value of its own, where an escape parts a word

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#527]
### What minimal example or steps are needed to reproduce the bug?

A dimension an escape welds to something Less compiles as a value of its own:

```less
a { width: 10px\#fff; }
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "overrides": [
    {
      "files": ["**/*.less"],
      "customSyntax": "postcss-less",
      "rules": {
        "@stylistic/less/unit-case": "upper"
      }
    }
  ]
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.less" --config myconfig.json
```

### What did you expect to happen?

Nothing said about `\#fff`. Less parts a word at an escape: `less@4.9.0` compiles that line to `a { width: 10px \#fff; }`, a dimension whose unit is `px` and, beside it, an escaped value it prints exactly as it was written. So `fff` is not part of the unit under this syntax, whatever it is under plain CSS.

### What actually happened?

```
1:12-19  Expected "px\#fff" to be "PX\#FFF"
```

and `--fix` leaves `a { width: 10PX\#FFF; }`, which Less compiles to `a { width: 10PX \#FFF; }` — the escaped value recased along with the unit.

The same holds for every other character an escape covers behind a dimension:

| input | option | what `--fix` leaves | what Less compiles it to |
| --- | --- | --- | --- |
| `a { width: 10px\#fff; }` | `upper` | `a { width: 10PX\#FFF; }` | `a { width: 10PX \#FFF; }` |
| `a { width: 10PX\@VAR; }` | `lower` | `a { width: 10px\@var; }` | `a { width: 10px \@var; }` |
| `a { width: 10PX\!important; }` | `lower` | `a { width: 10px\!important; }` | `a { width: 10px\!important; }` |

No file stops compiling: the escape is what keeps `\@VAR` from being a variable reference, and Less prints such a value literally. What changes is the case of a value Less carries through to the stylesheet it writes.

### What is the cause?

The reading is plain CSS's, and it is right there: `@csstools/css-tokenizer` reads `10px\#fff` as one dimension whose unit is `px#fff`, and `lightningcss@1.33.0` and Sass both print the word whole. `@stylistic/less/unit-case` inherits that reading, and Less is the one compiler of the three that parts the word.

### Do you have a proposal to fix the bug?

Give the Less syntax a reading of its own for a word an escape parts, so the unit ends where Less ends it.
